sections/featured-product.liquid

```diff
@@ -139,13 +139,21 @@
                     <div class="field">
                       <span id="ShareMessage-{{ section.id }}" class="share-button__message hidden" role="status">
                       </span>
+                      {%- liquid 
+                        assign share_link = product.url
+                        if product.selected_variant.url != blank
+                          assign share_link = product.selected_variant.url
+                        endif
+                      -%}
                       <input type="text"
                             class="field__input"
                             id="url"
-                            value="{{ shop.url | append: product.url }}"
+                            value="{{ shop.url | append: share_link }}"
                             placeholder="{{ 'general.share.share_url' | t }}"
                             onclick="this.select();"
                             readonly
+                            share-input
```

Comment:
This generates invalid HTML. I'd suggest applying a unique `id` and use this as the selector.